lib/Rx/Disposable/SerialDisposable.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Rx\Disposable;
+
+use Rx\DisposableInterface;
+
+class SerialDisposable implements DisposableInterface
+{
+    private $isDisposed = false;
+
+    /** @var DisposableInterface */
+    private $disposable = null;
+
+    function dispose()
+    {
+        if ($this->isDisposed) {
+            return;
+        }
+
+        $this->isDisposed = true;
+        $old = $this->disposable;
+        $this->disposable = null;
+
+        if ($old instanceof DisposableInterface) {
```

Comment:
Why this `instanceof` check? Could `setDisposable()` below get a typehint instead?